### PR TITLE
bpo-29884: faulthandler: Restore the old sigaltstack during teardown

### PR DESCRIPTION
**Pull request by tich**

### CODE REVIEW

**Review by vstinner — CHANGES_REQUESTED** [review]
The fix itself LGTM, but please also:

* Add yourself to Misc/ACKS
* Mention the fix in Misc/NEWS, in the Library section

**Comment by serhiy-storchaka** [comment]
And tests.

**Comment by vstinner** [comment]
> And tests.

IMHO it's overkill to test this bugfix. It's a corner case which requires to write a program which embeds Python and uses sigaltstack().

**Comment by tich** [comment]
Added myself to ACKS and listed the fix in NEWS. Thanks!

File: Misc/NEWS [hunk 1]
```diff
@@ -775,6 +775,9 @@ Library
 - Issue #29581: ABCMeta.__new__ now accepts **kwargs, allowing abstract base
   classes to use keyword parameters in __init_subclass__. Patch by Nate Soares.
 
+- Issue #29884: faulthandler: Restore the old sigaltstack during teardown.
```
Comment (serhiy-storchaka) [inline Misc/NEWS]: Move it to the start of the Library section.

Comment (tich) [reply Misc/NEWS]: Done

**Comment by serhiy-storchaka** [comment]
> IMHO it's overkill to test this bugfix. It's a corner case which requires to write a program which embeds Python and uses sigaltstack().

Okay, then tests are not required.

**Review by vstinner — APPROVED** [review]

**Comment by vstinner** [comment]
Thanks! I merged your fix. It would be nice to backport the fix Python 3.5 & 3.6 (using git cherry-pick).

**Pull request merged**